machineconfiguration/v1/types_machineconfignode.go

```diff
@@ -0,0 +1,266 @@
+	// The desired version represents the machine config the node will attempt to update to. This gets set before the machine config operator validates
+	// the new machine config against the current machine config.
+	// +required
+	ConfigVersion MachineConfigNodeSpecMachineConfigVersion `json:"configVersion"`
+
+	// pinnedImageSets holds the desired pinned image sets that this node should pin and pull.
+	// +listType=map
+	// +listMapKey=name
+	// +kubebuilder:validation:MaxItems=100
+	// +optional
+	PinnedImageSets []MachineConfigNodeSpecPinnedImageSet `json:"pinnedImageSets,omitempty"`
+}
+
+// MCOObjectReference holds information about an object the MCO either owns
+// or modifies in some way
+type MCOObjectReference struct {
+	// name is the object name.
+	// Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123) consisting
+	// of only lowercase alphanumeric characters, hyphens (-), and periods (.), start and end
+	// with an alphanumeric character, and be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:XValidation:rule="!format.dns1123Subdomain().validate(self).hasValue()",message="a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character."
+	// +required
+	Name string `json:"name"`
+}
+
+// MachineConfigNodeStatus holds the reported information on a particular machine config node.
+type MachineConfigNodeStatus struct {
+	// conditions represent the observations of a machine config node's current state.
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+	// observedGeneration represents the generation observed by the controller.
+	// This field is updated when the controller observes a change to the desiredConfig in the configVersion of the machine config node spec.
+	// +required
+	ObservedGeneration int64 `json:"observedGeneration"`
+	// configVersion describes the current and desired machine config for this node.
+	// The current version represents the current machine config for the node and is updated after a successful update.
+	// The desired version represents the machine config the node will attempt to update to.
+	// This desired machine config has been compared to the current machine config and has been validated by the machine config operator as one that is valid and that exists.
+	// +required
+	ConfigVersion MachineConfigNodeStatusMachineConfigVersion `json:"configVersion"`
+	// pinnedImageSets describes the current and desired pinned image sets for this node.
+	// The current version is the generation of the pinned image set that has most recently been successfully pulled and pinned on this node.
```

Comment:
These two lines of comments are a bit confusing. I think they are trying to describe some of the subfield's details. We should probably be a bit more general in describing what the status is holding